core/shared/platform/windows/win_time.c

```diff
@@ -18,3 +68,126 @@ os_time_get_boot_microsecond()
 
     return ((uint64)ts.tv_sec) * 1000 * 1000 + ((uint64)ts.tv_nsec) / 1000;
 }
+
+uint64
+os_win_clock_res_get(__wasi_clockid_t clock_id, __wasi_timestamp_t *resolution)
+{
+    switch (clock_id) {
+        case __WASI_CLOCK_MONOTONIC:
+        {
+             const __wasi_timestamp_t result =
+                (uint64_t)(((uint64_t)1000000000)
+                           / current_value_of_peformance_counter());
+            *resolution = (__wasi_timestamp_t)result;
+            __wasi_errno_t error = 0;
+            return __WASI_ESUCCESS;
+            
+        }
+        case __WASI_CLOCK_REALTIME:
+        case __WASI_CLOCK_PROCESS_CPUTIME_ID:
+        case __WASI_CLOCK_THREAD_CPUTIME_ID:
+        {
+            PULONG MaximumTime;
+            PULONG MinimumTime;
+            PULONG CurrentTime;
+            NTSTATUS
+            NTAPI Res;
+            if (Res = NtQueryTimerResolution(&MaximumTime, &MinimumTime,
+                                             &CurrentTime))
+                ;
```

Comment:
This doesn't seem to be correct; looks like whatever the result is, we'll ignore it, and always jump to line 98